### ISSUE TITLE
User-agent short codes not working on M1 Macbook

### ISSUE THREAD

**Issue by akash329d** [nativefier/nativefier#1211]
**Homework**

- [ x] I looked at `nativefier --help` and https://github.com/nativefier/nativefier/blob/master/API.md
- [ x] I checked the [Troubleshooting section of the README](https://github.com/nativefier/nativefier/blob/master/README.md#troubleshooting).
- [ x] I searched existing issues, open & closed. Yes, my bug is new.
- [ x] I'm using the latest version available at https://github.com/nativefier/nativefier/releases

**Bug description**

Running nativefier under arm64, the --user-agent shortcuts do not work, instead they are copied verbatim as the user agent. 

**Steps to reproduce**

Give your ***full*** nativefier command and its logs, with the ***`--verbose` flag***, on a ***public*** site:

```
nativefier 'https://www.whatismybrowser.com/detect/what-is-my-user-agent' --user-agent firefox
```

**Expected behavior**

What you expected to happen.

The user agent to be replaced with the firefox user agent.

**Actual behavior**

![image](https://user-images.githubusercontent.com/14977531/120728508-ef5bb980-c491-11eb-9f00-19408daaf753.png)

**Context**

 - Nativefier: (for example: 9.1.0) 43.1.3
 - Node.js: (for example: 14.6.0) v16.3.0
 - Npm: (for example: 6.14.7) 7.16.0
 - OS: (for example: Windows 10 build 1809) MacOS

**Comment by mattruzzi** [comment]
This seems to affect Intel Macs too. I can reproduce on my 2011 Intel Mac mini running macOS 10.13 High Sierra. 
```bash
❯ nativefier -v
43.1.3

~
❯ node -v
v16.3.0

~
❯ sw_vers
ProductName:	Mac OS X
ProductVersion:	10.13.6
BuildVersion:	17G14042

~

❯ nativefier 'https://www.whatismybrowser.com/detect/what-is-my-user-agent' --user-agent firefox --verbose

Running in verbose mode! This will produce a mountain of logs and is recommended only for troubleshooting or if you like Shakespeare.

Performing async options post-processing.
Inferring icon for https://www.whatismybrowser.com/detect/what-is-my-user-agent on darwin
Inferring icon from store for https://www.whatismybrowser.com/detect/what-is-my-user-agent on darwin
Allowed icon formats when building for darwin: [ '.icns', '.png', '.ico' ]
Inferring name for https://www.whatismybrowser.com/detect/what-is-my-user-agent
Got 249 icons from gitcloud
Max icon match score: 0
No relevant icon in store.
Trying to extract a .png icon from the page.
Fetched 26.8 kb page at https://www.whatismybrowser.com/detect/what-is-my-user-agent
Inferred title: What is my user agent? - WhatIsMyBrowser.com
Sanitized filename for What is my user agent? - WhatIsMyBrowser.com : What is my user agent - WhatIsMyBrowsercom
Got an icon from the page.
Writing 44.9 kb icon to /var/folders/x4/d6xhrn615_15s2bdvqp7rtw80000gr/T/nativefier-18-58-56-iconinfer--10113-ygElraAWiug5/icon.png
options {
  packager: {
    appCopyright: undefined,
    appVersion: undefined,
    arch: 'x64',
    asar: false,
    buildVersion: undefined,
    darwinDarkModeSupport: false,
    dir: '/Users/USER/.n/lib/node_modules/nativefier/app',
    electronVersion: '12.0.7',
    icon: '/var/folders/x4/d6xhrn615_15s2bdvqp7rtw80000gr/T/nativefier-18-58-56-iconinfer--10113-ygElraAWiug5/icon.png',
    name: 'What is my user agent - WhatIsMyBrowsercom',
    out: '/Users/USER',
    overwrite: true,
    platform: 'darwin',
    portable: false,
    targetUrl: 'https://www.whatismybrowser.com/detect/what-is-my-user-agent',
    tmpdir: false,
    upgrade: false,
    upgradeFrom: undefined,
    win32metadata: {
      ProductName: undefined,
      InternalName: undefined,
      FileDescription: undefined
    }
  },
  nativefier: {
    accessibilityPrompt: true,
    alwaysOnTop: false,
    backgroundColor: null,
    basicAuthPassword: null,
    basicAuthUsername: null,
    blockExternalUrls: false,
    bookmarksMenu: null,
    bounce: false,
    browserwindowOptions: undefined,
    clearCache: false,
    counter: false,
    crashReporter: undefined,
    disableContextMenu: undefined,
    disableDevTools: undefined,
    disableGpu: false,
    diskCacheSize: null,
    disableOldBuildWarning: false,
    enableEs3Apis: false,
    fastQuit: false,
    fileDownloadOptions: undefined,
    flashPluginDir: null,
    fullScreen: false,
    globalShortcuts: null,
    hideWindowFrame: undefined,
    ignoreCertificate: false,
    ignoreGpuBlacklist: false,
    inject: [],
    insecure: false,
    internalUrls: null,
    lang: undefined,
    maximize: false,
    nativefierVersion: '43.1.3',
    processEnvs: undefined,
    proxyRules: null,
    showMenuBar: false,
    singleInstance: false,
    titleBarStyle: null,
    tray: false,
    userAgent: 'firefox',
    userAgentOverriden: true,
    verbose: true,
    versionString: undefined,
    width: 1280,
    height: 800,
    minWidth: undefined,
    minHeight: undefined,
    maxWidth: undefined,
    maxHeight: undefined,
    widevine: false,
    x: undefined,
    y: undefined,
    zoom: 1
  }
}

Preparing Electron app...
Copying electron app from /Users/USER/.n/lib/node_modules/nativefier/app to /var/folders/x4/d6xhrn615_15s2bdvqp7rtw80000gr/T/nativefier-18-58-56-app--10113-l3FJdp1WbAbB
Writing app config to /var/folders/x4/d6xhrn615_15s2bdvqp7rtw80000gr/T/nativefier-18-58-56-app--10113-l3FJdp1WbAbB/nativefier.json
No files to inject, skipping copy.
Updating /var/folders/x4/d6xhrn615_15s2bdvqp7rtw80000gr/T/nativefier-18-58-56-app--10113-l3FJdp1WbAbB/package.json 'name' field to what-is-my-user-agent---whatismybrowsercom-nativefier-2b6c22

Converting icons...
Converting icon /var/folders/x4/d6xhrn615_15s2bdvqp7rtw80000gr/T/nativefier-18-58-56-iconinfer--10113-ygElraAWiug5/icon.png to /var/folders/x4/d6xhrn615_15s2bdvqp7rtw80000gr/T/nativefier-18-58-56-iconconv--10113-IWZpe8XCxzuh/icon.icns. Calling shell command: "/Users/USER/.n/lib/node_modules/nativefier/icon-scripts/convertToIcns" "/var/folders/x4/d6xhrn615_15s2bdvqp7rtw80000gr/T/nativefier-18-58-56-iconinfer--10113-ygElraAWiug5/icon.png" "/var/folders/x4/d6xhrn615_15s2bdvqp7rtw80000gr/T/nativefier-18-58-56-iconconv--10113-IWZpe8XCxzuh/icon.icns"
Conversion succeeded and produced icon at /var/folders/x4/d6xhrn615_15s2bdvqp7rtw80000gr/T/nativefier-18-58-56-iconconv--10113-IWZpe8XCxzuh/icon.icns
Copying icons if necessary
No copying necessary on macOS; aborting

Packaging... This will take a few seconds, maybe minutes if the requested Electron isn't cached yet...
  electron-packager Electron Packager 15.2.0
  electron-packager Node v16.3.0
  electron-packager Host Operating system: darwin 17.7.0 (x64) +0ms
  electron-packager Packager Options: {"arch":"x64","asar":false,"darwinDarkModeSupport":false,"dir":"/var/folders/x4/d6xhrn615_15s2bdvqp7rtw80000gr/T/nativefier-18-58-56-app--10113-l3FJdp1WbAbB","electronVersion":"12.0.7","icon":"/var/folders/x4/d6xhrn615_15s2bdvqp7rtw80000gr/T/nativefier-18-58-56-iconconv--10113-IWZpe8XCxzuh/icon.icns","name":"What is my user agent - WhatIsMyBrowsercom","out":"/Users/USER","overwrite":true,"platform":"darwin","portable":false,"targetUrl":"https://www.whatismybrowser.com/detect/what-is-my-user-agent","tmpdir":false,"upgrade":false,"win32metadata":{}} +1ms
  electron-packager Target Platforms: darwin +1ms
  electron-packager Target Architectures: x64 +0ms
  electron-packager Inferring appVersion from version in /var/folders/x4/d6xhrn615_15s2bdvqp7rtw80000gr/T/nativefier-18-58-56-app--10113-l3FJdp1WbAbB/package.json +0ms
  electron-packager Application name: What is my user agent - WhatIsMyBrowsercom +25ms
  electron-packager Target Electron version: 12.0.7 +0ms
  electron-packager Ignored path regular expressions: [
  '/package-lock\\.json$',
  '/yarn\\.lock$',
  '/\\.git($|/)',
  '/node_modules/\\.bin($|/)',
  '\\.o(bj)?$'
] +0ms
  electron-packager Downloading Electron with options {"platform":"darwin","arch":"x64","version":"12.0.7","artifactName":"electron"} +0ms
Packaging app for platform darwin x64 using electron v12.0.7
  electron-packager Creating /Users/USER/darwin-x64-template +6ms
  electron-packager Extracting /Users/USER/Library/Caches/electron/4f029753a9a8576806dced687079cdf26638acb9ef0f41141be4eff390c1bd52/electron-v12.0.7-darwin-x64.zip to /Users/USER/darwin-x64-template +60ms
  electron-packager Initializing app in /Users/USER/What is my user agent - WhatIsMyBrowsercom-darwin-x64 from /Users/USER/darwin-x64-template template +0ms
  electron-packager Ignored paths based on the out param: [
  '/Users/USER/What is my user agent - WhatIsMyBrowsercom-darwin-x64',
  '/Users/USER/What is my user agent - WhatIsMyBrowsercom-darwin-arm64',
  '/Users/USER/What is my user agent - WhatIsMyBrowsercom-linux-ia32',
  '/Users/USER/What is my user agent - WhatIsMyBrowsercom-linux-x64',
  '/Users/USER/What is my user agent - WhatIsMyBrowsercom-linux-armv7l',
  '/Users/USER/What is my user agent - WhatIsMyBrowsercom-linux-arm64',
  '/Users/USER/What is my user agent - WhatIsMyBrowsercom-linux-mips64el',
  '/Users/USER/What is my user agent - WhatIsMyBrowsercom-mas-x64',
  '/Users/USER/What is my user agent - WhatIsMyBrowsercom-mas-arm64',
  '/Users/USER/What is my user agent - WhatIsMyBrowsercom-win32-ia32',
  '/Users/USER/What is my user agent - WhatIsMyBrowsercom-win32-x64',
  '/Users/USER/What is my user agent - WhatIsMyBrowsercom-win32-arm64'
] +7s
  electron-packager Copying icon "/var/folders/x4/d6xhrn615_15s2bdvqp7rtw80000gr/T/nativefier-18-58-56-iconconv--10113-IWZpe8XCxzuh/icon.icns" to app's Resources as "electron.icns" +0ms
  electron-packager Renaming Electron to What is my user agent - WhatIsMyBrowsercom in /Users/USER/What is my user agent - WhatIsMyBrowsercom-darwin-x64/Electron.app/Contents/MacOS +233ms
  electron-packager Renaming Electron Helper to What is my user agent - WhatIsMyBrowsercom Helper in /Users/USER/What is my user agent - WhatIsMyBrowsercom-darwin-x64/Electron.app/Contents/Frameworks/Electron Helper.app/Contents/MacOS +2ms
  electron-packager Renaming Electron Helper (Renderer) to What is my user agent - WhatIsMyBrowsercom Helper (Renderer) in /Users/USER/What is my user agent - WhatIsMyBrowsercom-darwin-x64/Electron.app/Contents/Frameworks/Electron Helper (Renderer).app/Contents/MacOS +0ms
  electron-packager Renaming Electron Helper (Plugin) to What is my user agent - WhatIsMyBrowsercom Helper (Plugin) in /Users/USER/What is my user agent - WhatIsMyBrowsercom-darwin-x64/Electron.app/Contents/Frameworks/Electron Helper (Plugin).app/Contents/MacOS +0ms
  electron-packager Renaming Electron Helper (GPU) to What is my user agent - WhatIsMyBrowsercom Helper (GPU) in /Users/USER/What is my user agent - WhatIsMyBrowsercom-darwin-x64/Electron.app/Contents/Frameworks/Electron Helper (GPU).app/Contents/MacOS +1ms
  electron-packager Renaming Electron Helper.app to What is my user agent - WhatIsMyBrowsercom Helper.app in /Users/USER/What is my user agent - WhatIsMyBrowsercom-darwin-x64/Electron.app/Contents/Frameworks +0ms
  electron-packager Renaming Electron Helper (Renderer).app to What is my user agent - WhatIsMyBrowsercom Helper (Renderer).app in /Users/USER/What is my user agent - WhatIsMyBrowsercom-darwin-x64/Electron.app/Contents/Frameworks +1ms
  electron-packager Renaming Electron Helper (Plugin).app to What is my user agent - WhatIsMyBrowsercom Helper (Plugin).app in /Users/USER/What is my user agent - WhatIsMyBrowsercom-darwin-x64/Electron.app/Contents/Frameworks +0ms
  electron-packager Renaming Electron Helper (GPU).app to What is my user agent - WhatIsMyBrowsercom Helper (GPU).app in /Users/USER/What is my user agent - WhatIsMyBrowsercom-darwin-x64/Electron.app/Contents/Frameworks +1ms

Finalizing build...
App built to /Users/USER/What is my user agent - WhatIsMyBrowsercom-darwin-x64, move to wherever it makes sense for you and run the app bundle.
```

**Comment by TheCleric** [comment]
I apologize for the issue here. THis has been updated in our docs, but hasn't made it out to npm yet. Hopefully very soon!

**Comment by ronjouch** [comment]
> I apologize for the issue here. THis has been updated in our docs, but hasn't made it out to npm yet. Hopefully very soon!

Truuuuuuuuuuue. Will remind to release fast when we merge stuff that updates the docs! Cutting a release later today, sorry for the hiccup everybody!

**Comment by ronjouch** [comment]
Say hi to useragent shortcodes in freshly-released 44.0.0, released now on npm! Thanks for the report. With that, closing, feel free to re-open if it's not working as intended.
